src/Development/IDE/Core/Shake.hs

```diff
@@ -346,6 +348,11 @@ shakeOpen getLspId eventer withProgress withIndefiniteProgress logger debouncer
         publishedDiagnostics <- newVar mempty
         positionMapping <- newVar HMap.empty
         let restartShakeSession = shakeRestart ideState
+        progressQueue <- newTVarIO KickCompleted
```

Comment:
It’s a bit surprising that this is calle `Queue` but then it’s just a `TVar` instead of a queue.

Reply:
good point - it was originally a queue but then I realized only the most recent value matters. Will change.